### ISSUE TITLE
bug(material/input): No focus indication when readonly input gains focus

### ISSUE THREAD

**Issue by zelliott** [angular/components#22783]
#### Reproduction

Stackblitz: https://stackblitz.com/edit/angular-zci3aa?file=src/app/form-field-appearance-example.html
 
#### Expected Behavior

Some focus indication on the readonly inputs.

#### Actual Behavior

No focus indication on the readonly inputs.

#### Environment

- Angular: Latest
- CDK/Material: Latest
- Browser(s): All
- Operating System (e.g. Windows, macOS, Ubuntu): All

#### Discussion

This appears to have been purposefully removed in https://github.com/angular/components/pull/7273 to make Material readonly inputs consistent with native readonly inputs. However, native readonly inputs now appear to render focus indicators on focus (at least on Chrome, FF), tested on https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly#result.

From a WCAG compliance POV, this is _likely_ a [2.4.7 Focus Visible](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html) risk as there doesn't appear to be any exception made for readonly inputs. The fix may be as easy as dropping the `(!this.readonly || !isFocused)` condition on https://github.com/angular/components/blob/0a4dbe1706e33cbf14b6c65fc9e364cf04277625/src/material/input/input.ts#L347.
